service/history/ndc/workflow_state_replicator.go

```diff
@@ -621,6 +621,7 @@ func (r *WorkflowStateReplicatorImpl) getNewRunMutableState(
 	newRunID string,
 	originalMutableState workflow.MutableState,
 	newRunEventsBlob *commonpb.DataBlob,
+	isStateBased bool,
```

Comment:
nit: I think workflow_state_replicator only has new run in state based stack.

Reply:
yeah, my bad. My comment lead to this change. But I think it's fine to keep as is unless you have strong opinion.